prov/terraform/openstack/osinstance.go

```diff
@@ -251,147 +377,192 @@ func (g *osGenerator) generateOSInstance(ctx context.Context, kv *api.KV, cfg co
+	outputs[path.Join(instancesKey, instanceName, "/attributes/public_ip_address")] = publicIPKey
 
-		id, err := deployments.LookupInstanceAttributeValue(ctx, kv, deploymentID, serverGroup, "0", "id")
-		if err != nil {
-			return err
-		}
+	return nil
+}
 
-		compute.SchedulerHints.Group = id
-		return nil
+func computeNetworkAttributes(ctx context.Context, opts osInstanceOptions,
+	networkNodeName, instancesKey string,
+	instance *ComputeInstance, outputs map[string]string) error {
+
+	log.Debugf("Looking for Network id for %q", networkNodeName)
+	var networkID string
+	resultChan := make(chan string, 1)
+	go func() {
+		for {
+			nID, err := deployments.GetInstanceAttributeValue(
+				opts.kv, opts.deploymentID, networkNodeName, opts.instanceName, "network_id")
+			if err != nil {
+				log.Printf("[Warning] bypassing error while waiting for a network id: %v", err)
+			}
+			// As networkID is an optional property GetInstanceAttribute then GetProperty
+			// may return an empty networkID so keep checking as long as we have it
+			if nID != nil && nID.RawString() != "" {
+				resultChan <- nID.RawString()
+				return
+			}
+			select {
+			case <-time.After(1 * time.Second):
+			case <-ctx.Done():
+				// context cancelled, give up!
+				return
+			}
+		}
+	}()
+	select {
+	case networkID = <-resultChan:
+	case <-ctx.Done():
+		return ctx.Err()
+	}
+	cn := ComputeNetwork{UUID: networkID, AccessNetwork: false}
+	i := len(instance.Networks)
+	if instance.Networks == nil {
+		instance.Networks = make([]ComputeNetwork, 0)
 	}
+	instance.Networks = append(instance.Networks, cn)
+
+	// Provide output for network_name, network_id, addresses attributes
+	networkIDKey := opts.nodeName + "-" + opts.instanceName + "-networkID"
+	networkNameKey := opts.nodeName + "-" + opts.instanceName + "-networkName"
+	networkAddressesKey := opts.nodeName + "-" + opts.instanceName + "-addresses"
+	commons.AddOutput(opts.infrastructure, networkIDKey, &commons.Output{
+		Value: fmt.Sprintf("${%s.%s.network.%d.uuid}",
+			opts.resourceTypes[computeInstance], instance.Name, i)})
+	commons.AddOutput(opts.infrastructure, networkNameKey, &commons.Output{
+		Value: fmt.Sprintf("${%s.%s.network.%d.name}",
+			opts.resourceTypes[computeInstance], instance.Name, i)})
+	commons.AddOutput(opts.infrastructure, networkAddressesKey, &commons.Output{
+		Value: fmt.Sprintf("[ ${%s.%s.network.%d.fixed_ip_v4} ]",
+			opts.resourceTypes[computeInstance], instance.Name, i)})
+
+	prefix := path.Join(instancesKey, opts.instanceName, "attributes/networks", strconv.Itoa(i))
+	outputs[path.Join(prefix, "network_name")] = networkNameKey
+	outputs[path.Join(prefix, "network_id")] = networkIDKey
+	outputs[path.Join(prefix, "addresses")] = networkAddressesKey
+	return nil
+}
 
-	return errors.Errorf("Only one group requirement can be accepted for OpenStack compute with name:%q", nodeName)
+func getProperyValueString(kv *api.KV, deploymentID, nodeName, propertyName string) (string, error) {
```

Comment:
typo on func name. Maybe, you can move this func into deployments package, nodes.go file ?

Reply:
ok, will change this

Reply:
re_used deployments.GetStringNodeProperty()